src/MongoQP/QueryProfiler.php

```diff
@@ -25,7 +25,13 @@ function($database) { return $database['name']; },
 
     public function getCollections($database)
     {
-        return $this->mongo->selectDB($database)->getCollectionNames();
+        $mongodb = $this->mongo->selectDB($database);
+
+        if (!$mongodb->getSlaveOkay()) {
+              $mongodb->setSlaveOkay();
+        }
```

Comment:
[`setSlaveOkay()`](http://php.net/manual/en/mongodb.setslaveokay.php) has been deprecated since 1.2.x. It would be wiser to call [`setReadPreference()`](http://php.net/manual/en/mongodb.setreadpreference.php) with `MongoClient::RP_PRIMARY_PREFERRED` instead.